bluebox/wave.py

```diff
@@ -0,0 +1,88 @@
+"""wave.py
+
+This file contains the Wave class for generating waveforms.
+It is completely generic and backend agnostic, but does not
+provide any functionality for playing the waveforms. That is
+handled in the backend specific files.
+"""
+
+import typing as t
+import math
+from array import array
+
+
+class SineWave:
+    """SineWave class for generating waveform arrays."""
+
+    _sample_rate: float = 44100.0
+    _channels: int = 1
+
+    def __init__(
+                self,
+                sr: t.Optional[float] = None,
+                ch: t.Optional[int] = None) -> None:
+        """Initialize the Wave object."""
+        if sr is not None:
+            self._sample_rate = sr
+        if ch is not None:
+            self._channels = ch
+
+    def sine(
+            self,
+            freq: float,
+            length: float,
+            amplitude: float = 1.0,
+            phase: float = 0.0) -> t.MutableSequence[float]:
```

Comment:
Instead of returning the entire tone, make this a generator too, then it can be compatible with streams or backends that require bytes

Reply:
E.g. without any effort you can wrap it and return an int, byte or whatever version of each point...it should be pretty efficient (at least memory wise)